### PR TITLE
0x11-V6, 0x07-V2, 0x09-V4, 0x08-V3

### PR DESCRIPTION
**Pull request by MotinhaAna**
Thank you for submitting a Pull Request to the Mobile Application Security Verification Standard. Please make sure that:

- [ ] Your PR is linked to an issue.
- [ ] Your PR follows the [contribution guidelines](https://github.com/OWASP/owasp-masvs/blob/master/CONTRIBUTING.md "Contribution guidelines")
- [ ] All links are valid and use HTTPS and have been formatted as [TEXT](URL “NAME”)
- [ ] In case of a translation: have you made sure that the translation from EN is as close to the EN document as possible?

This PR covers issue #< insert number here >.

If your pull-request has a focus of rewriting a requirement and/or adding a new one, please make sure you have a backing issue in which your change has been discussed and agreed. If there is none, then your PR will be closed without further review.

### CODE REVIEW

File: Document-ptpt/0x08-V3-Cryptography_Verification_Requirements.md [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
 
-Cryptography is an essential ingredient when it comes to protecting data stored on a mobile device. It is also a category where things can go horribly wrong, especially when standard conventions are not followed. The purpose of the controls in this chapter is to ensure that the verified application uses cryptography according to industry best practices, including:
+A criptografia é um conjunto de práticas/ protocolos essenciais no processo de protecção de dados em dispositivos móveis. Quando esses protocolos não são correctamente implementados, tornam-se pouco seguros e possivelmente exploráveis. O propósito deste capítulo é assegurar a aplicação das boas práticas da criptografia, o que inclui:
 
-- Use of proven cryptographic libraries;
-- Proper choice and configuration of cryptographic primitives;
-- A suitable random number generator wherever randomness is required.
+- O uso de bibliotecas de criptografia verificadas;
+- A configuração correta das primitivas criptográficas;
+- O uso de um gerador de números aleatórios adequado.
```
Comment (luisfontes19) [inline Document-ptpt/0x08-V3-Cryptography_Verification_Requirements.md:9]: "... quando aleatoriedade é necessaria"

File: Document-ptpt/0x08-V3-Cryptography_Verification_Requirements.md [hunk 2]
```diff
@@ -1,32 +1,32 @@
...
-| **3.2** | MSTG-CRYPTO-2 | The app uses proven implementations of cryptographic primitives. | ✓ | ✓ |
-| **3.3** | MSTG-CRYPTO-3 | The app uses cryptographic primitives that are appropriate for the particular use-case, configured with parameters that adhere to industry best practices. | ✓ | ✓ |
-| **3.4** | MSTG-CRYPTO-4 | The app does not use cryptographic protocols or algorithms that are widely considered deprecated for security purposes. | ✓ | ✓ |
-| **3.5** | MSTG-CRYPTO-5 | The app doesn't re-use the same cryptographic key for multiple purposes. | ✓ | ✓ |
-| **3.6** | MSTG-CRYPTO-6 | All random values are generated using a sufficiently secure random number generator. | ✓ | ✓ |
+| **3.1** | MSTG-CRYPTO-1 | A aplicação não se baseia em criptografia simétrica com chaves escritas em código imutável (hardcoded) como método único de encriptação. | ✓ | ✓ |
+| **3.2** | MSTG-CRYPTO-2 | A aplicação faz uso de primitivas criptográficas verificadas. | ✓ | ✓ |
+| **3.3** | MSTG-CRYPTO-3 | A aplicação faz uso de primitivas criptográficas apropriadas ao caso específico, configuradas de acordo com as boas práticas. | ✓ | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x08-V3-Cryptography_Verification_Requirements.md:17]: configuradas com parametros que seguem as melhores praticas da industria

File: Document-ptpt/0x08-V3-Cryptography_Verification_Requirements.md [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
+| **3.4** | MSTG-CRYPTO-4 | A aplicação não faz uso de protocolos ou algoritmos criptográficos que são considerados descontinuados. | ✓ | ✓ |
+| **3.5** | MSTG-CRYPTO-5 | A aplicação não faz uso da mesma chave para múltiplos propósitos. | ✓ | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x08-V3-Cryptography_Verification_Requirements.md:19]: da mesma chave criptografica

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 4]
```diff
@@ -1,35 +1,36 @@
-# V4: Authentication and Session Management Requirements
+# V4: Autenticação e Requisitos para a Gestão da Sessão
 
-## Control Objective
+## Objectivo
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:3]: I'm using "Objetivo de Controlo"
We should choose one and use it consistently across files

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 5]
```diff
@@ -1,35 +1,36 @@
...
 
-In most cases, users logging into a remote service is an integral part of the overall mobile app architecture. Even though most of the logic happens at the endpoint, MASVS defines some basic requirements regarding how user accounts and sessions are to be managed.
+De acordo com a arquitectura da grande maioria das aplicações, a autenticação dos utilizadores é realizada num serviço remoto. Ainda que a maior parte da lógica aplicacional seja feita no ponto de acesso remoto, MASVS define alguns requisitos relativos à gestão da sessão e da conta dos utilizadores..
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:5]: "is an integral part of the overall mobile app architecture" is not translated

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 6]
```diff
@@ -1,35 +1,36 @@
...
-| **4.12** | MSTG-AUTH-12 | Authorization models should be defined and enforced at the remote endpoint. | ✓ | ✓ |
-
-## References
-
-The OWASP Mobile Security Testing Guide provides detailed instructions for verifying the requirements listed above.
+| **4.1** | MSTG-AUTH-1 | Se a aplicação providencia ao utilizador acesso a um serviço remoto, é necessário garantir a autenticação do utilizador no ponto de acesso remoto, a partir do nome de utilizador e respectiva palavra-passe, ou de alguma outra forma válida. | ✓ | ✓ |
+| **4.2** | MSTG-AUTH-2 | Se a gestão da sessão é feita com persistência de estado, o acesso remoto deve fazer uso de identificador de sessão gerado aleatoriamente, para garantir a autenticação do cliente/utilizador a cada pedido ao servidor, sem que seja necessário o repetitivo fornecimento das credenciais do utilizador. | ✓ | ✓ |
+| **4.3** | MSTG-AUTH-3 | Se a gestão da sessão é baseada num passe (token), sem persistência de estado, o servidor providencia o passe assinado através de um algoritmo seguro. | ✓ | ✓ |
+| **4.4** | MSTG-AUTH-4 | O ponto de acesso remoto termina a existente sessão quando o utilizador sai da aplicação (logout). | ✓ | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:15]: existente sessão -> "sessão existente" sounds better

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 7]
```diff
@@ -1,35 +1,36 @@
...
+| **4.5** | MSTG-AUTH-5 | Deve ser definida uma colecção de regras para a definição da palavra-passe, e é necessário garantir que essas regras são cumpridas a nível do ponto de acesso/servidor. | ✓ | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:16]: you can abreviate " colecção de regras para a definição da palavra-passe" to "politica"

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 8]
```diff
@@ -1,35 +1,36 @@
...
+| **4.6** | MSTG-AUTH-6 | O ponto de acesso / servidor deve implementar um mecanismo que permita proteger a aplicação de uma submissão excessiva de tentativas de autenticação. | ✓ | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:17]: loose the '/ servidor'

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 9]
```diff
@@ -1,35 +1,36 @@
...
+| **4.7** | MSTG-AUTH-7 | As sessões devem ser invalidadas ao nível do endpoint / servidor após um período de inactividade pré-definido, ao fim do qual, os passes de acesso (access token) expiram. | ✓ | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:18]: endpoint -> ponto de acesso

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 10]
```diff
@@ -1,35 +1,36 @@
...
+| **4.8** | MSTG-AUTH-8 | A autenticação biométrica, se aplicável, não deve ser implementada através de uma API que retorna Verdadeiro ou Falso. Deve ser baseada no desbloqueio de um gestor de credenciais (keychain/keystore). | | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:19]: try to make it more similar to the original

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 11]
```diff
@@ -1,35 +1,36 @@
...
+| **4.9** | MSTG-AUTH-9 | Existe uma autenticação reforçada (second factor authentication, 2FA) no ponto de acesso / servidor e esse modo de autenticação é reforçado.  | | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:20]: loose the '/ servidor'

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 12]
```diff
@@ -1,35 +1,36 @@
...
+| **4.10** | MSTG-AUTH-10 | A transmissão de informação sensível deve ser feita após o reforço da autenticação. | | ✓ |
+| **4.11** | MSTG-AUTH-11 | A aplicação informa o utilizador de qualquer actividade sensível efectuada com a sua conta. Aos utilizadores é providenciada uma lista de dispositivos onde a sua conta está a ser utilizada, assim como outras informações contextuais (localização, identificador respectivo à conexão da internet, entre outras). É também providenciado o bloqueamento dessas mesmas sessões. | | ✓ |
```
Comment (luisfontes19) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:22]: try to make it more similar to the original

File: Document-ptpt/0x11-V6-Interaction_with_the_environment.md [hunk 13]
```diff
@@ -1,33 +1,33 @@
-# V6: Platform Interaction Requirements
+# V6: Requisitos para Interacção com a plataforma
 
-## Control Objective
+## Objectivo
 
-The controls in this group ensure that the app uses platform APIs and standard components in a secure manner. Additionally, the controls cover communication between apps (IPC).
+Os controlos neste grupo asseguram que a aplicação utiliza Interfaces de Programação da Aplicação (APIs) e componentes estandardizadas. Adicionalmente é também abordada a comunicação entre aplicações (ICP)..
```
Comment (luisfontes19) [inline Document-ptpt/0x11-V6-Interaction_with_the_environment.md:5]: estandardizadas is pt-br, go with padronizadas

File: Document-ptpt/0x07-V2-Data_Storage_and_Privacy_requirements.md [hunk 14]
```diff
@@ -1,49 +1,50 @@
...
+| **2.11** | MSTG-STORAGE-11 | A aplicação reforça uma política mínima de segurança no acesso ao dispositivo, como por exemplo a necessidade do utilizador para definir um código-passe. |  | ✓ |
+| **2.12** | MSTG-STORAGE-12 | A aplicação indica ao utilizador os tipos de informação privada processada, assim como as boas práticas a seguir durante a utilização da mesma. |  | ✓ |
+| **2.13** | MSTG-STORAGE-13 | Nenhuma informação sensível é guardada localmente no dispositivo móvel. Esses mesmos dados são obtidos através de um pedido a um servidor externo quando necessário, sendo mantidos em memória enquanto necessário. |  | ✓ |
+| **2.14** | MSTG-STORAGE-14 | Caso seja necessário guardar localmente informação sensível, esta deve ser encriptada com uma chave proveniente de um dispositivo de armazenamento que exija autenticação para o acesso ao mesmo. |  | ✓ |
+| **2.15** | MSTG-STORAGE-15 | O armazenamento local da aplicação deve ser limpo após uma excessiva quantidade de tentativas de autenticaçãos. |  | ✓ |
```
Comment (filipagcx) [inline Document-ptpt/0x07-V2-Data_Storage_and_Privacy_requirements.md:38]: tentativas de **autenticação**.

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md (hunk 5 above)
Comment (filipagcx) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:5]: ponto final a mais no final do parágrafo

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md [hunk 15]
```diff
@@ -1,35 +1,36 @@
...
-| **4.10** | MSTG-AUTH-10 | Sensitive transactions require step-up authentication. | | ✓ |
-| **4.11** | MSTG-AUTH-11 | The app informs the user of all sensitive activities with their account. Users are able to view a list of devices, view contextual information (IP address, location, etc.), and to block specific devices. | | ✓ |
-| **4.12** | MSTG-AUTH-12 | Authorization models should be defined and enforced at the remote endpoint. | ✓ | ✓ |
-
-## References
-
-The OWASP Mobile Security Testing Guide provides detailed instructions for verifying the requirements listed above.
+| **4.1** | MSTG-AUTH-1 | Se a aplicação providencia ao utilizador acesso a um serviço remoto, é necessário garantir a autenticação do utilizador no ponto de acesso remoto, a partir do nome de utilizador e respectiva palavra-passe, ou de alguma outra forma válida. | ✓ | ✓ |
+| **4.2** | MSTG-AUTH-2 | Se a gestão da sessão é feita com persistência de estado, o acesso remoto deve fazer uso de identificador de sessão gerado aleatoriamente, para garantir a autenticação do cliente/utilizador a cada pedido ao servidor, sem que seja necessário o repetitivo fornecimento das credenciais do utilizador. | ✓ | ✓ |
```
Comment (filipagcx) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:13]: deve fazer uso **do** identificador de sessão

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md (hunk 11 above)
Comment (filipagcx) [inline Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:20]: Existe uma segunda autenticação (second factor authentication, 2FA) no ponto de acesso / servidor e esse modo de autenticação é reforçado.

File: Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md (hunk 12 above)
Comment (filipagcx) [reply Document-ptpt/0x09-V4-Authentication_and_Session_Management_Requirements.md:22]: **bloqueio** dessas mesmas sessões

File: Document-ptpt/0x11-V6-Interaction_with_the_environment.md (hunk 13 above)
Comment (filipagcx) [reply Document-ptpt/0x11-V6-Interaction_with_the_environment.md:5]: ponto final a mais no final do parágrafo

File: Document-ptpt/0x11-V6-Interaction_with_the_environment.md [hunk 16]
```diff
@@ -1,33 +1,33 @@
...
+| **6.5** | MSTG-PLATFORM-5 | JavaScript está inactivo nas vistas da aplicação excepto nos casos em que é estritamente necessário. | ✓ | ✓ |
+| **6.6** | MSTG-PLATFORM-6 | As vistas da aplicação estão configuradas de forma a suportar o número mínimo de protocolos , idealmente só https deve ser suportado.  Potentially dangerous handlers, such as file, tel and app-id, are disabled. | ✓ | ✓ |
+| **6.7** | MSTG-PLATFORM-7 | No caso de métodos nativos da aplicação estarem expostos na vista, verificar que esta apenas renderiza o JavaScript contido na aplicação. | ✓ | ✓ |
+| **6.8** | MSTG-PLATFORM-8 | A deserialização de objectos, se necessária, é implementada através do uso de interfaces (APIs) de serialização seguras. | ✓ | ✓ |
+| **6.9** | MSTG-PLATFORM-9 | A aplicação protege-se contra ataques de sobreposição de ecrãs. (Apenas em Android) |  | ✓ |
+| **6.10** | MSTG-PLATFORM-10 | A memória da vista, o armazenamento da mesma e os recursos utilizados (JavaScript, etc.) devem ser limpas após a destruição da vista. |  | ✓ |
```
Comment (filipagcx) [inline Document-ptpt/0x11-V6-Interaction_with_the_environment.md:20]: A memória da vista **web**, o armazenamento da mesma e os recursos utilizados (JavaScript, etc.) devem ser **limpos** após a destruição da vista **web**.

File: Document-ptpt/0x11-V6-Interaction_with_the_environment.md [hunk 17]
```diff
@@ -1,33 +1,33 @@
-# V6: Platform Interaction Requirements
+# V6: Requisitos para Interacção com a plataforma
 
-## Control Objective
+## Objectivo
```
Comment (filipagcx) [inline Document-ptpt/0x11-V6-Interaction_with_the_environment.md:3]:
```suggestion
## Objetivo de Controlo
```

**Comment by luisfontes19** [comment]
@MotinhaAna  please delete this PR as you created a new one with the changes

**Comment by sushi2k** [comment]
Thanks @luisfontes19. If that's the case I will close this PR.

**Pull request closed without merging**